### PR TITLE
Adjust for joint waypoint joint name order

### PR DESCRIPTION
**Pull request by Levi-Armstrong**
Since joint waypoint order is not guaranteed to match the manipulator the planners must adjust the waypoint data to match the manipulator.

### CODE REVIEW

**Comment by Levi-Armstrong** [comment]
@drchrislewis Make the change now

**Comment by Levi-Armstrong** [comment]
@drchrislewis I added another commit to check when the start joint waypoint does not match the start of the seed trajectory which causes trajopt to never converge.

File: tesseract/tesseract_planning/tesseract_motion_planners/src/trajopt/config/trajopt_planner_default_config.cpp [hunk 1]
```diff
@@ -88,21 +88,40 @@ std::shared_ptr<trajopt::ProblemConstructionInfo> TrajOptPlannerDefaultConfig::g
...
+    // If it does not start this causes issues in trajopt and it will never converge.
+    if (isJointWaypointType(target_waypoints.front()->getType()))
+    {
+      const auto jwp = std::static_pointer_cast<JointWaypoint>(target_waypoints.front());
+      const Eigen::VectorXd position = jwp->getPositions(kin->getJointNames());
+      for (int i = 0; i < static_cast<int>(kin->numJoints()); ++i)
+      {
+        if (std::abs(position[i] - seed_trajectory(0, i)) > static_cast<double>(std::numeric_limits<float>::epsilon()))
+        {
+          CONSOLE_BRIDGE_logError("Seed trajectory start position does not match starting joint waypoint position!");
```
Comment (drchrislewis) [inline tesseract/tesseract_planning/tesseract_motion_planners/src/trajopt/config/trajopt_planner_default_config.cpp]: I'd prefer the logged error indicated the joint that failed. Then you'd have to go through every one but return false if one or more failed.

Comment (Levi-Armstrong) [reply tesseract/tesseract_planning/tesseract_motion_planners/src/trajopt/config/trajopt_planner_default_config.cpp]: Fixing it now.

File: tesseract/tesseract_planning/tesseract_motion_planners/src/trajopt/config/trajopt_planner_default_config.cpp [hunk 2]
```diff
@@ -88,21 +88,44 @@ std::shared_ptr<trajopt::ProblemConstructionInfo> TrajOptPlannerDefaultConfig::g
...
+      const Eigen::VectorXd position = jwp->getPositions(kin->getJointNames());
+      for (int i = 0; i < static_cast<int>(kin->numJoints()); ++i)
+      {
+        if (std::abs(position[i] - seed_trajectory(0, i)) > static_cast<double>(std::numeric_limits<float>::epsilon()))
+        {
+          std::stringstream ss;
+          ss << "Seed trajectory start position does not match starting joint waypoint position!";
+          ss << "    waypoint: " << position.transpose().matrix() << std::endl;
+          ss << "  seed start: " << seed_trajectory.row(0).transpose().matrix() << std::endl;
+          CONSOLE_BRIDGE_logError(ss.str().c_str());
```
Comment (drchrislewis) [inline tesseract/tesseract_planning/tesseract_motion_planners/src/trajopt/config/trajopt_planner_default_config.cpp]: @Levi-Armstrong Looks great.

**Pull request merged**